sdk/tests/SerializationAndDeserializationTests.cc

```diff
@@ -0,0 +1,199 @@
+  EXPECT_EQ(protoContractId->realmnum(), testRealmNum);
+  EXPECT_EQ(protoContractId->contractnum(), testContractNum);
+}
+
+TEST_F(SerializationAndDeserializationTests, DeserializeContractIdFromProtobufTest)
+{
+  // Given
+  uint64_t testShardNum = getTestShardNum();
+  uint64_t testRealmNum = getTestRealmNum();
+  uint64_t testContractNum = getTestContractNum();
+  
+  proto::ContractID testProtoContractId = proto::ContractID();
+  testProtoContractId.set_shardnum(testShardNum);
+  testProtoContractId.set_realmnum(testRealmNum);
+  testProtoContractId.set_contractnum(testContractNum);
+
+  // When
+  ContractId contractId = ContractId::fromProtobuf(testProtoContractId);
+  
+  // Then
+  EXPECT_EQ(contractId.getShardNum(), testShardNum);
+  EXPECT_EQ(contractId.getRealmNum(), testRealmNum);
+  EXPECT_EQ(contractId.getContractNum(), testContractNum);
+}
+
+TEST_F(SerializationAndDeserializationTests, SerializeTransactionIdToProtobufTest)
+{
+  // Given
+  const std::string testAccountIdStr = "111.222.333";
+  const std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+  AccountId testAccountId(testAccountIdStr);
+  TransactionId testTransactionId = TransactionId::generate(testAccountId);
+
+  // When
+  const auto protoTransactionIdPtr = std::unique_ptr<proto::TransactionID>(testTransactionId.toProtobuf());
+  const auto protoTimestampPtr = std::unique_ptr<proto::Timestamp>(internal::TimestampConverter::toProtobuf(now));
+
+  // Then
+  EXPECT_EQ(static_cast<uint64_t>(protoTransactionIdPtr->accountid().shardnum()), testAccountId.getShardNum());
+  EXPECT_EQ(static_cast<uint64_t>(protoTransactionIdPtr->accountid().realmnum()), testAccountId.getRealmNum());
+  EXPECT_EQ(static_cast<uint64_t>(protoTransactionIdPtr->accountid().accountnum()), testAccountId.getAccountNum());
+  EXPECT_EQ(protoTransactionIdPtr->transactionvalidstart().seconds(), protoTimestampPtr->seconds());
+}
+
+TEST_F(SerializationAndDeserializationTests, DeserializeTransactionIdFromProtobufTest)
+{
+  // Given
+  const std::string testAccountIdStr = "111.222.333";
+  const std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+  AccountId testAccountId(testAccountIdStr);
+  proto::TransactionID testProtoTransactionId;
+  testProtoTransactionId.set_allocated_accountid(testAccountId.toProtobuf().release());
+  testProtoTransactionId.set_allocated_transactionvalidstart(internal::TimestampConverter::toProtobuf(now));
+
+  // When
+  const TransactionId transactionId = TransactionId::fromProtobuf(testProtoTransactionId);
+
+  // Then
+  EXPECT_EQ(transactionId.getAccountId().getAccountNum(), testAccountId.getAccountNum());
+  EXPECT_EQ(transactionId.getAccountId().getRealmNum(), testAccountId.getRealmNum());
+  EXPECT_EQ(transactionId.getAccountId().getShardNum(), testAccountId.getShardNum());
+  EXPECT_EQ(transactionId.getValidTransactionTime(), now);
+}
+
+TEST_F(SerializationAndDeserializationTests, SerializeTransferToProtobufTest)
```

Comment:
To write this test I used test [ProtoTransferTest](https://github.com/hashgraph/hedera-sdk-cpp/blob/c54139e8bc8f507e1f08d052b15eec3d188a0bd0/sdk/tests/TransferTest.cc#L33) from [TransferTest.cc](https://github.com/hashgraph/hedera-sdk-cpp/blob/main/sdk/tests/TransferTest.cc).

Reply:
Test `SerializeTransferToProtobufTest` into `TransferTest.cc` in commit https://github.com/hashgraph/hedera-sdk-cpp/pull/176/commits/abcec567906e347c29c9a94934ce4f14d47963d5.